modules/svg/mapilio_images.js

```diff
@@ -7,30 +7,107 @@ import {svgPath, svgPointTransform} from './helpers';
 
 export function svgMapilioImages(projection, context, dispatch) {
     const throttledRedraw = _throttle(function () { dispatch.call('change'); }, 1000);
-    const minZoom = 12;
+    const imageMinZoom = 16;
+    const lineMinZoom = 10;
+    const viewFieldZoomLevel = 18;
     let layer = d3_select(null);
     let _mapilio;
-    const viewFieldZoomLevel = 18;
-
+    let _viewerYaw = 0;
+    let _activeUsernameFilter;
+    let _activeIds;
 
     function init() {
         if (svgMapilioImages.initialized) return;
         svgMapilioImages.enabled = false;
         svgMapilioImages.initialized = true;
     }
 
-
     function getService() {
         if (services.mapilio && !_mapilio) {
             _mapilio = services.mapilio;
-            _mapilio.event.on('loadedImages', throttledRedraw);
+            _mapilio.event
+                .on('loadedImages', throttledRedraw)
+                .on('loadedLines', throttledRedraw);
         } else if (!services.mapilio && _mapilio) {
             _mapilio = null;
         }
 
         return _mapilio;
     }
 
+    /**
+     * Filters images
+     * @param {*} images
+     * @returns array of filtered images
+     */
+    async function filterImages(images) {
+        var fromDate = context.photos().fromDate();
+        var toDate = context.photos().toDate();
+        const username = context.photos().usernames();
+
+        const service = getService();
+
+        if (fromDate) {
+            images = images.filter(function(image) {
+                return new Date(image.capture_time).getTime() >= new Date(fromDate).getTime();
+            });
+        }
+        if (toDate) {
+            images = images.filter(function(image) {
+                return new Date(image.capture_time).getTime() <= new Date(toDate).getTime();
+            });
+        }
+        if (username && service) {
+            if (_activeUsernameFilter !== username) {
+                _activeUsernameFilter = username;
+
+                const tempIds = await service.getUserIds(username);
+
+                _activeIds = {};
+                tempIds.forEach(id => _activeIds[id] = true);
```

Comment:
when a username filter is set, this crashes:

![image](https://github.com/user-attachments/assets/142cf5f2-bb9e-4861-8fa5-75147a759f50)



I'm not sure if the mapilio API officially supports user information in the public API. I found the endpoint `https://end.mapilio.com/api/search-user?options[parameters][keywords]=…` which kind of works, but a) the search is fuzzy and b) the endpoint is not included in the API documentation at https://docs.mapilio.com/